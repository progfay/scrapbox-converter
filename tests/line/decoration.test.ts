/* global describe it expect */

import { parse } from '@progfay/scrapbox-parser'
import convert from '../../src'
import pretty from 'pretty'

describe('decoration', () => {
  it('Simple decoration', () => {
    const input = `Decoration
[*** deco-*-3]`
    const html = convert(parse(input), 'projectName')
    expect(pretty(html)).toMatchSnapshot()
  })

  it('Multi decoration', () => {
    const input = `Decoration
[/!_- deco-! deco-_ deco-- deco-/]`
    const html = convert(parse(input), 'projectName')
    expect(pretty(html)).toMatchSnapshot()
  })
})
