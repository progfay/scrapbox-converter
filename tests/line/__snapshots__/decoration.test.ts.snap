// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`decoration Multi decoration 1`] = `
"<article>
  <h1 class=\\"title\\">Decoration</h1>
  <section>
    <p><strong class=\\"deco-/ deco-! deco-_ deco--\\">deco-!&nbsp;deco-_&nbsp;deco--&nbsp;deco-/</strong></p>
  </section>
</article>"
`;

exports[`decoration Simple decoration 1`] = `
"<article>
  <h1 class=\\"title\\">Decoration</h1>
  <section>
    <p><strong class=\\"deco-*-3\\">deco-*-3</strong></p>
  </section>
</article>"
`;
